File: src/main/java/codereview/simpleorder/presentation/item/ItemController.java
package codereview.simpleorder.presentation.item;

import codereview.simpleorder.application.item.ItemService;
import codereview.simpleorder.dto.item.CreateItemRequest;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

@RestController
@RequiredArgsConstructor
@RequestMapping("/items")
public class ItemController {

    private final ItemService itemService;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public Long registerItem(@RequestBody CreateItemRequest request) {

        Long savedItem = itemService.registerItem(request);

        return savedItem;
    }
}

File: item/src/main/kotlin/msa/with/ddd/item/ui/ItemController.kt
package msa.with.ddd.item.ui

import msa.with.ddd.item.application.ItemService
import msa.with.ddd.item.dto.item.ItemCreateRequest
import msa.with.ddd.item.dto.item.ItemResponses
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RequestMapping("/items")
@RestController
class ItemController(private val itemService: ItemService) {

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun add(@RequestBody request: ItemCreateRequest): Long {
        return itemService.registerItem(request)
    }

    @GetMapping
    fun list(@RequestParam(value = "ids", required = false) itemIds: List<Long>?): ItemResponses {
        val items = itemService.findItems(itemIds)
        return ItemResponses(items)
    }
}